### ISSUE TITLE
Problem with Telegram after cleaning/optimizing databases

### ISSUE THREAD

**Issue by sam-kzn** [d4rken/sdmaid-public#1542]
Hi!
After cleaning and optimizing databases, the Telegram app starts to work incorrectly.
Chat list loading gets loop.

**Comment by d4rken** [comment]
Some more info please: Device, Android Version, Root, SD Maid version

Is SD Maid on default settings? "Skip running apps" can affect this.
Some apps have issues if they get launched while SD Maid is working on their database.
Could this have happened here?

Can you record a logcat to see if the Telegram app prints errors?

**Comment by sam-kzn** [comment]
Device: Highscreen Boost 3 SE Pro
OS: Android 6.0
GAPPS: Deleted (using MicroG)
Root: MagiskSU

SD Maid Settings:
**Application settings:**
"Allow system apps" - On
"Show unavailable elements" - On
"Skip runnig apps" - Off
**Database settings:**
"Skip runnig apps" - On

SD Maid Manager shows that Telegram app is stoped.
Telegram doesn' print any errors.
Just very slow (∞) loading list of chats.

**Comment by d4rken** [comment]
> SD Maid Manager shows that Telegram app is stoped.

Stopped as in brown tag color STOPPED? A forced stopped app can not start itself this would mean that it could not have launched itself while SD Maid was working on the database...

Is this reproducable? How do you fix it? If you run the databases tool again, does the issue reoccur?

**Comment by sam-kzn** [comment]
"Stopped" - turquoise color (between green and blue) (it was stopped by double clicking to "Backward" button).
Yes, it's reproducable every time when I'm using SD Maid app features.
So, You can reproduce it Yourself (https://telegram.org/).
Just add several chats and groups in the app.

**Comment by d4rken** [comment]
I can't reproduce this on my Pixel@8.1.0.

The telegram databases itself can't be `VACUUM`ed by SD Maid  anyways due to incompatible structures. SD Maid will attempt it, then abort on error. I've tried it with both Telegram running and in the `STOPPED` state.

The only way I can slightly (I don't have many chats) delay launch is by using the AppCleaner on Telegram, but then a slower launch is expected if caches have to be rebuild.

Can you record a video of how you reproduce it?
Telegram no issue -> opening SD Maid, doing something -> Telegram issue
If you don't want to post the video publicly (due to Telegram) you can also mail it to be via support@darken.eu.

**Comment by sam-kzn** [comment]
Good Day!
I've send the video to Your e-mail.

**Comment by d4rken** [comment]
Can you narrow down which tool causes this? I'm not sure if it is the databases tool, it could also be the AppCleaner tool.

If you just force-stop telegram through SD Maid, then open it again, is everything still okay?

**Comment by sam-kzn** [comment]
Yes, after force-stop the Telelgram app through the SD Maid, it's still work properly.
I've tested each of SD Maid tool separatd.
The problem arises from the Database tool.
Video was send to Your e-mail.

**Comment by d4rken** [comment]
I've seen the video, I just saw it deleting many Telegram files in the AppCleaner too, which is why I asked that.

So we narrowed it down to the Database tool. The VACUUM operation executed by it is not something I can change.

The question now is how to proceed.

Options:
* Global (removable) default exclusion for Telegram from the databases tool?
* Leave it as it is any people who have issues with this should manually exclude telegram?
* Telegram has many databases, find out if it is a specific one and only exclude that one?

What do you think?
Do you know if this is a common problem?

**Comment by sam-kzn** [comment]
May be the problem arises from database encrypting by local password?

**Comment by d4rken** [comment]
I've searched a bit but could not find any indication that [VACUUM](https://sqlite.org/lang_vacuum.html) and encryption are problematic.

**Comment by sam-kzn** [comment]
I've tryed to use the Databases tool on my wife's device.
The Telegram app still works properly with her account. But she has a much few chats.
Today I'll try to login in to the Telegram with my account and try the test again.

**Comment by d4rken** [comment]
Do you see a difference in files? If it's a specific file that would a clean way to fix it if we don't have to exclude all Telegram files.

> The Telegram app still works properly with her account. But she has a much few chats.

This would explain why I could not reproduce it (i don't use telegram so I only had few chats too).

**Comment by sam-kzn** [comment]
Hello!
Finally I've tested the DataBase Tool with my Telegram account on my wife's device. The problem doesn't reproduce on it.
But from my device the problem is gone too.
I don't know reason of it.

**Comment by sam-kzn** [comment]
> Did you have the same amount of chats?
Yes, after adding account the chats and groups synchronizing 1:1.

> Do you think an exclusion should be added anyways?
I'll test it for several weeks for final answer. Ok?

**Comment by d4rken** [comment]
> I'll test it for several weeks for final answer. Ok?

👍
